java/vector/src/main/codegen/templates/AbstractPromotableFieldWriter.java

```diff
@@ -144,6 +174,16 @@ public ListWriter list() {
     return getWriter(MinorType.LIST).list();
   }
 
+  @Override
+  public MapWriter map() {
+    return getWriter(MinorType.LIST).map();
+  }
+
+  @Override
+  public MapWriter map(boolean keysSorted) {
+    return getWriter(MinorType.MAP, new ArrowType.Map(keysSorted));
```

Comment:
Do we need to have two static final instaces of `ArrowType.Map` that corresponds to cases when keysSorted equals to `true` and `false`?
This may avoid some object allocations.

Reply:
This would probably make the most sense, I'm just not sure it should be addressed in this PR or as a separate issue.